nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/test/java/org/apache/nifi/processors/slack/PutSlackTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.slack;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class PutSlackTest {
+
+    private TestRunner testRunner;
+    public static String WEBHOOK_TEST_URL = "https://hooks.slack.com/services/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";
```

Comment:
The Webhook URL property is marked as sensitive, but there's one here in the test. It appears that the unit tests are not actually posting to a Slack (since they test error conditions), but even so the URL should be something guaranteed to work without the internet or a particular site being available. I see the URL being used in the integration test as well, to actually post to what appears to be a real Slack. I'm thinking this should be mocked in the unit tests and stubbed (with a fake web-server perhaps) in the integration tests